lib/new_relic/agent/aws.rb

```diff
@@ -5,13 +5,64 @@
 module NewRelic
   module Agent
     module Aws
-      def self.create_arn(service, resource, region)
-        return unless NewRelic::Agent.config[:'cloud.aws.account_id']
+      CHARACTERS = %w[A B C D E F G H I J K L M N O P Q R S T U V W X Y Z 2 3 4 5 6 7].freeze
+      HEX_MASK = '7fffffffff80'
 
-        "arn:aws:#{service}:#{region}:#{NewRelic::Agent.config[:'cloud.aws.account_id']}:#{resource}"
+      def self.create_arn(service, resource, region, account_id)
+        # if any of the values are nil, we can't create an ARN
+        return unless service && resource && region && account_id
```

Comment:
Do we want to log if we can't create the arn because of a `nil`

Reply:
Those are great points. The customer also wouldn't have anything they could do to fix the problem and start creating an ARN (outside of adding it via the config option, but that's easy for us to diagnose). Most of this data either comes from our instrumentation or from the AWS gem. I'm hesitant to give a customer logs that they can't do much about.

What if we keep this place in mind if we encounter problems with ARNs not getting created, and add logging here on a per-customer basis for GTSEs?

Reply:
Perfect - all sounds good to me too!